sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UnconditionalJumpStatement.cs

```diff
@@ -93,13 +70,95 @@ protected override void Initialize(SonarAnalysisContext context)
             (statement as ReturnStatementSyntax)?.ReturnKeyword.ToString() ??
             (statement as ThrowStatementSyntax)?.ThrowKeyword.ToString();
 
-        private static bool IsWithinMethodPropertyOrLambda(SyntaxNode node) =>
-            !node.IsAnyKind(methodOrPropertyDeclarations);
+        private class FooWalker : CSharpSyntaxWalker

```

Comment:
Does this walker support nested loops? Something like:

    while (true)
    {
        while (true)
        {
            if (something) continue;
        }
        break;
    }